packages/core/src/app/payment/Payment.tsx

```diff
@@ -547,15 +547,15 @@ class Payment extends Component<
 
     private async loadPaymentMethodsOrThrow(): Promise<void> {
         const {
-            defaultMethod,
             loadPaymentMethods,
             onUnhandledError = noop,
         } = this.props;
-        const { selectedMethod = defaultMethod } = this.state;
 
         try {
             await loadPaymentMethods();
 
+            const selectedMethod = this.state.selectedMethod || this.props.defaultMethod;
```

Comment:
We get `selectedMethod` and `defaultMethod` from `this.state` and `this.props` directly because they may change after async `loadPaymentMethods` method call

Reply:
This seems fine for now, do you think there is an opportunity to refactor this in future?

Reply:
We tried to move this implementation inside tracking method, but it looks a bit complicated. I guess it might be fixed when Payment component will be updated from class component to functional